jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererUtil.kt

```diff
@@ -196,18 +195,6 @@ object CodeWhispererUtil {
         listOf(CodeWhispererSsoLearnMoreAction(), ConnectWithAwsToContinueActionError(), DoNotShowAgainActionError())
     )
 
-    fun isAccessTokenExpired(project: Project): Boolean {
-        val tokenProvider = tokenProvider(project) ?: return false
```

Comment:
I think `?:` case here should return `true` instead of `false`

Reply:
previous we were not checking `isAccessTokenExpired`, which is wrong

Reply:
that's why we've observed cases where client registration, refresh token are still valid but customers are seeing InvalidGrant

Reply:
can we add tests here, I think this has been touched recently and it seems there's a regression.

Reply:
yep, agree, will add them.